### PR TITLE
MAINT: Relax tolerance on OSX only

### LINKED ISSUE #5158: Maybe tolerance error for test_zero_constrained, macOS
On wheel testing:

```
=================================== FAILURES ===================================
______________ TestGenericNegativeBinomial.test_zero_constrained _______________
../venv/lib/python2.7/site-packages/statsmodels/base/tests/test_generic_methods.py:220: in test_zero_constrained
    assert_allclose(res1.resid, res2.resid, rtol=1e-10)
E   AssertionError: 
E   Not equal to tolerance rtol=1e-10, atol=0
E   
E   (mismatch 0.302129767211%)
E    x: array([-3.174188, -1.174188, -3.174188, ...,  5.713472,  5.686109,
E           3.730759])
E    y: array([-3.174188, -1.174188, -3.174188, ...,  5.713472,  5.686109,
E           3.730759])
```

Example:

https://travis-ci.org/MacPython/statsmodels-wheels/jobs/426380464

Build matrix:

https://travis-ci.org/MacPython/statsmodels-wheels/builds/426380455

### PR DESCRIPTION
**Pull request by bashtage**
Relax tolerance on test that randomly fails on OpenBlas/OSX

- [X] closes #5158
- [X] code/documentation is well formatted.  
- [X] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy/dev/gitwash/development_workflow.html#writing-the-commit-message).

### CODE REVIEW

File: statsmodels/compat/_testing.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import platform
+
+__all__ = ['IS_OSX']
```
Comment (jbrockmendel) [inline statsmodels/compat/_testing.py]: Does this need it’s own module?

File: statsmodels/base/tests/test_generic_methods.py [hunk 2]
```diff
@@ -112,7 +113,12 @@ def test_zero_constrained(self):
         if hasattr(res1, 'resid'):
             # discrete models, Logit don't have `resid` yet
             # atol discussion at gh-5158
-            assert_allclose(res1.resid, res2.resid, rtol=1e-10, atol=1e-12)
+            rtol = 1e-10
+            atol = 1e-12
+            if IS_OSX:
+                rtol = 1e-8
```
Comment (jbrockmendel) [inline statsmodels/base/tests/test_generic_methods.py]: Comment pointing back to this PR?

File: statsmodels/compat/_testing.py (hunk 1 above)
Comment (bashtage) [reply statsmodels/compat/_testing.py]: It is now used in 2 distinct files, so I say yes.  It also prevents all of the `HAVE_MATPLOTLIB` style imports should this need arise again.

**Comment by bashtage** [comment]
This is only for azure, and so passing.

File: statsmodels/regression/tests/test_lme.py [hunk 3]
```diff
@@ -86,6 +88,8 @@ def f(x):
 class TestMixedLM(object):
 
     # Test analytic scores and Hessian using numeric differentiation
+    @pytest.mark.skipif(PLATFORM_OSX, reason='known to fail on OSX due to '
+                                             'numerical differences')
```
Comment (jbrockmendel) [inline statsmodels/regression/tests/test_lme.py]: shouldnt this be an xfail?  the test is failing for poorly-understood reasons, not invalid on OSX.

**Pull request merged**